private/pkg/git/cloner.go

```diff
@@ -160,10 +161,11 @@ func (c *cloner) CloneToBucket(
 		command.RunWithDir(baseDir.AbsPath()),
 	); err != nil {
 		// If the ref fetch failed, without a fallback, return the error.
-		if fallbackRef == "" || !strings.Contains(buffer.String(), "couldn't find remote ref") {
+		if fallbackRef == "" {
```

Comment:
This error check works for branches but fails on `HEAD~`. I've removed the error check and now always try the fallback if available.